### PR TITLE
feat: added 404 page and refactor error boundary page

### PR DESCRIPTION
**Pull request by isaackps**
# In this PR: 
- Updated the react router to handle wrong path, and redirected to 404 page 
Screenshot 2021-03-05 at 4 32 08 PM

### CODE REVIEW

File: src/components/ErrorBoundary/ErrorBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    this.setState({
+      hasError: true,
+    });
+  }
+
+  render() {
+    if (this.state.hasError) {
+      return (
+        <ErrorPage
+          title="Something went wrong!"
```
Comment (simboonlong) [inline src/components/ErrorBoundary/ErrorBoundary.tsx]:
```suggestion
          title="Something went wrong"
```

File: src/components/ErrorBoundary/ErrorPage/ErrorPage.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export default {
+  title: "Error/ErrorPage",
+  component: ErrorPage,
+  parameters: {
+    componentSubtitle: "General error page",
+  },
+};
+
+export const Default = () => {
+  return <ErrorPage title="ERROR!!" description="Something went wrong!" />;
```
Comment (simboonlong) [inline src/components/ErrorBoundary/ErrorPage/ErrorPage.stories.tsx]:
```suggestion
  return <ErrorPage title="Error" description="Something went wrong" />;
```
sounds too scary

File: src/components/ErrorBoundary/ErrorPage/ErrorPage.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        <img className="mx-auto" src="/static/images/errorpage/error.png" alt="error_img" style={{ height: "15vh" }} />
+      </div>
+      <h2 className="text-orange uppercase pt-5 pb-2 text-3xl md:text-4xl lg:text-5xl font-semibold">{title}</h2>
+      <p className="text-black pb-2 text-base md:text-lg lg:text-2xl">{description}</p>
+      <a
+        className="mt-4 inline-block px-8 py-4 bg-navy hover:bg-orange text-white hover:text-white border-none rounded-full font-semibold uppercase no-underline transition duration-300 ease-out text-sm"
+        href="/"
+      >
+        Go back to home
+      </a>
```
Comment (simboonlong) [inline src/components/ErrorBoundary/ErrorPage/ErrorPage.tsx]: use `Link`? `import { Link } from "react-router-dom";`

File: src/pages/page404.tsx [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import React from "react";
+import { ErrorPage } from "../components/ErrorBoundary/ErrorPage";
+
+export const Page404 = () => {
+  return <ErrorPage title="404" description="Oops, page not found" />;
```
Comment (simboonlong) [inline src/pages/page404.tsx]:
```suggestion
export const PageNotFound = () => {
  return <ErrorPage title="404" description="Oops, page not found" />;
```

minor nitpick on naming to more descriptive

File: src/routes.tsx [hunk 5]
```diff
@@ -2,8 +2,9 @@ import { EmailSuccessPage } from "./pages/emailSuccess";
 import { FaqPage } from "./pages/faq";
 import { HomePageContainer } from "./pages/home";
 import { MediaPage } from "./pages/media";
+import { Page404 } from "./pages/page404";
 import { ResourcesPage } from "./pages/resources";
```
Comment (simboonlong) [inline src/routes.tsx]:
```suggestion
import { PageNotFound } from "./pages/pageNotFound";
import { ResourcesPage } from "./pages/resources";
```

File: src/routes.tsx [hunk 6]
```diff
@@ -16,4 +17,5 @@ export const routes = [
   { path: "/email/success", exact: true, component: EmailSuccessPage },
   { path: "/resources", exact: true, component: ResourcesPage },
   { path: "/media", exact: true, component: MediaPage },
+  { path: "*", component: Page404 },
 ];
```
Comment (simboonlong) [inline src/routes.tsx]:
```suggestion
  { path: "*", component: PageNotFound },
];
```

**Comment by simboonlong** [comment]
not sure if we should catch all and redirect them to a dedicated `/404` path, dunno if this will affect the open in new tab nested documents not. i guess this approach is ok for now

File: src/components/ErrorBoundary/ErrorBoundary.tsx (hunk 1 above)
Comment (isaackps) [reply src/components/ErrorBoundary/ErrorBoundary.tsx]: ok

File: src/components/ErrorBoundary/ErrorPage/ErrorPage.stories.tsx (hunk 2 above)
Comment (isaackps) [reply src/components/ErrorBoundary/ErrorPage/ErrorPage.stories.tsx]: ok

File: src/pages/page404.tsx (hunk 4 above)
Comment (isaackps) [reply src/pages/page404.tsx]: ok

File: src/routes.tsx (hunk 5 above)
Comment (isaackps) [reply src/routes.tsx]: ok

File: src/routes.tsx (hunk 6 above)
Comment (isaackps) [reply src/routes.tsx]: ok

File: src/components/ErrorBoundary/ErrorPage/ErrorPage.tsx (hunk 3 above)
Comment (isaackps) [reply src/components/ErrorBoundary/ErrorPage/ErrorPage.tsx]: ok

**Review by simboonlong — APPROVED** [review]
lgtm

**Pull request merged**